packages/dd-trace/src/llmobs/span_processor.js

```diff
@@ -0,0 +1,195 @@
+'use strict'
+
+const {
+  SPAN_KIND,
+  MODEL_NAME,
+  MODEL_PROVIDER,
+  METADATA,
+  INPUT_MESSAGES,
+  INPUT_VALUE,
+  OUTPUT_MESSAGES,
+  INPUT_DOCUMENTS,
+  OUTPUT_DOCUMENTS,
+  OUTPUT_VALUE,
+  METRICS,
+  ML_APP,
+  TAGS,
+  PARENT_ID_KEY,
+  SESSION_ID,
+  NAME,
+  UNSERIALIZABLE_VALUE_TEXT
+} = require('./constants')
+
+const {
+  ERROR_MESSAGE,
+  ERROR_TYPE,
+  ERROR_STACK
+} = require('../constants')
+
+const AgentlessWriter = require('./writers/spans/agentless')
+const AgentProxyWriter = require('./writers/spans/agentProxy')
+const { isLLMSpan } = require('./util')
+
+const tracerVersion = require('../../../../package.json').version
+const logger = require('../log')
+
+class LLMObsSpanProcessor {
+  constructor (config) {
+    this._config = config
+    const { llmobs } = config
+
+    if (llmobs.enabled) {
+      const LLMObsSpanWriter = llmobs.agentlessEnabled ? AgentlessWriter : AgentProxyWriter
+      this._writer = new LLMObsSpanWriter(config)
+    }
+  }
+
+  process (span) {
+    if (!this._config.llmobs.enabled) return
+    if (!isLLMSpan(span)) return
```

Comment:
nit: maybe rename this to isLLMObsSpan to make it clear we're checking for LLM Obs spans, and not the llm span kind